tests/scripts/thread-cert/Cert_5_1_01_RouterAttach.py

```diff
@@ -168,6 +179,317 @@ def test(self):
+            filter_coap_ack(ADDR_SOL_URI).\
+            filter(lambda p: {\
+                              NL_STATUS_TLV,\
+                              NL_RLOC16_TLV,\
+                              NL_ROUTER_MASK_TLV\
+                              } == set(p.coap.tlv.type) and\
+                   p.coap.code == COAP_CODE_ACK and\
+                   p.thread_address.tlv.status == 0\
+                   ).\
+            must_next()
+
+        # Step 8: Router Sends a Link Request Message.
+        #         The Link Request Message MUST be multicast and contain
+        #         the following TLVs:
+        #             - Challenge TLV
+        #             - Leader Data TLV
+        #             - Source Address TLV
+        #             - Version TLV
+        #             - TLV Request TLV: Link Margin
+
+        pkts.filter_wpan_src64(ROUTER).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_LINK_REQUEST).\
+            filter(lambda p: {\
+                              CHALLENGE_TLV,\
+                              LEADER_DATA_TLV,\
+                              SOURCE_ADDRESS_TLV,\
+                              VERSION_TLV,\
+                              TLV_REQUEST_TLV,\
+                              LINK_MARGIN_TLV\
+                              } <= set(p.mle.tlv.type)\
+                   ).\
+            must_next()
+
+        # Step 9: Leader sends a Unicast Link Accept and Request Message.
+        #         The Message MUST be unicast to Router
+        #         The Message MUST contain the following TLVs:
+        #             - Leader Data TLV
+        #             - Link-layer Frame Counter TLV
+        #             - Link Margin TLV
+        #             - Response TLV
+        #             - Source Address TLV
+        #             - Version TLV
+        #             - Challenge TLV (optional)
+        #             - MLE Frame Counter TLV (optional)
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_wpan_dst64(ROUTER).\
+            filter_mle_cmd(MLE_LINK_ACCEPT_AND_REQUEST).\
+            filter(lambda p: {\
+                              LEADER_DATA_TLV,\
+                              LINK_LAYER_FRAME_COUNTER_TLV,\
+                              LINK_MARGIN_TLV,\
+                              RESPONSE_TLV,\
+                              SOURCE_ADDRESS_TLV,\
+                              VERSION_TLV,\
+                              CHALLENGE_TLV,\
+                              MLE_FRAME_COUNTER_TLV\
+                              } <= set(p.mle.tlv.type) or\
+                             {\
+                              LEADER_DATA_TLV,\
+                              LINK_LAYER_FRAME_COUNTER_TLV,\
+                              LINK_MARGIN_TLV,\
+                              RESPONSE_TLV,\
+                              SOURCE_ADDRESS_TLV,\
+                              VERSION_TLV\
+                               } <= set(p.mle.tlv.type)).\
```

Comment:
```suggestion
            filter(lambda p: {\
                              LEADER_DATA_TLV,\
                              LINK_LAYER_FRAME_COUNTER_TLV,\
                              LINK_MARGIN_TLV,\
                              RESPONSE_TLV,\
                              SOURCE_ADDRESS_TLV,\
                              VERSION_TLV\
                               } <= set(p.mle.tlv.type)).\
```

These two code is same.